modules/videopress/utility-functions.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * Validate user-supplied guid values against expected inputs
+ *
+ * @since 1.1
+ * @param string $guid video identifier
+ * @return bool true if passes validation test
+ */
+function videopress_is_valid_guid( $guid ) {
+	if ( ! empty( $guid ) && strlen( $guid ) === 8 && ctype_alnum( $guid ) ) {
+		return true;
+	}
+	return false;
+}
+
+/**
+ * Get details about a specific video by GUID:
+ *
+ * @param $guid string
+ * @return object
+ */
+function videopress_get_video_details( $guid ) {
+	if ( ! videopress_is_valid_guid( $guid ) ) {
+		return new WP_Error( 'bad-guid-format', __( 'Invalid Video GUID!', 'jetpack' ) );
+	}
+
+	$version  = '1.1';
+	$endpoint = sprintf( '/videos/%1$s', $guid );
+	$response = wp_remote_get( sprintf( 'https://public-api.wordpress.com/rest/v%1$s%2$s', $version, $endpoint ) );
+	$data     = json_decode( wp_remote_retrieve_body( $response ) );
+
+	/**
+	 * Allow functions to modify fetched video details.
+	 *
+	 * This filter allows third-party code to modify the return data
+	 * about a given video.  It may involve swapping some data out or
+	 * adding new parameters.
+	 *
+	 * @since 3.10
+	 *
+	 * @param object $data The data returned by the WPCOM API. See: https://developer.wordpress.com/docs/api/1.1/get/videos/%24guid/
+	 * @param string $guid The GUID of the VideoPress video in question.
+	 */
+	return apply_filters( 'videopress_get_video_details', $data, $guid );
+}
+
+
+/**
+ * Get an attachment ID given a URL.
+ *
+ * Modified from http://wpscholar.com/blog/get-attachment-id-from-wp-image-url/
+ *
+ * @todo: Add some caching in here.
+ *
+ * @param string $url
+ *
+ * @return int|bool Attachment ID on success, false on failure
+ */
+function videopress_get_attachment_id_by_url( $url ) {
+	// Strip out protocols, so it doesn't fail because searching for http: in https: dir.
+	$dir = set_url_scheme( trailingslashit( wp_upload_dir()['baseurl'] ), 'relative' );
```

Comment:
`foo()['bar']` 
[FAD](https://wiki.php.net/rfc/functionarraydereferencing) was introduced in 5.4 :(